simphony/tests/test_hash.py

```diff
@@ -0,0 +1,108 @@
+# Copyright © Simphony Project Contributors
+# Licensed under the terms of the MIT License
+# (see simphony/__init__.py for details)
+
+from simphony.libraries import siepic
+from simphony.libraries import sipann
+
+
+class TestSiepicHash:
+
+    def test_hash(self):
+
+        bdc1 = siepic.BidirectionalCoupler(thickness=2.2e-7, width=5e-7)
+        bdc2 = siepic.BidirectionalCoupler(thickness=2.2e-7, width=5e-7)
+        assert (bdc1.__hash__() == bdc2.__hash__())
+
+        bdc1 = siepic.BidirectionalCoupler(2.1e-7)
+        bdc2 = siepic.BidirectionalCoupler(2.3e-7)
+        assert (bdc1.__hash__() != bdc2.__hash__())
+
+        hr1siepic = siepic.HalfRing(gap=1e-7, radius=1e-5, width=5e-7, thickness=2.2e-7)
+        hr2siepic = siepic.HalfRing(gap=1e-7, radius=1e-5, width=5e-7, thickness=2.2e-7)
+        assert (hr1siepic.__hash__() == hr2siepic.__hash__())
+
+        hr1siepic = siepic.HalfRing(gap=8e-8, radius=10e-6, width=5.2e-7, thickness=2.1e-7)
+        hr2siepic = siepic.HalfRing(gap=1e-7, radius=1e-5, width=5e-7, thickness=2.2e-7)
```

Comment:
For components with multiple parameters, you should probably do multiple tests where each test only has 1 parameter different. That way you can get full coverage.